diff -Naur linux-3.2.32.o/include/linux/netfilter/xt_owner.h linux-3.2.32/include/linux/netfilter/xt_owner.h
--- linux-3.2.32.o/include/linux/netfilter/xt_owner.h	2012-10-17 10:50:15.000000000 +0800
+++ linux-3.2.32/include/linux/netfilter/xt_owner.h	2015-03-02 22:15:10.363105657 +0700
@@ -7,6 +7,7 @@
 	XT_OWNER_UID    = 1 << 0,
 	XT_OWNER_GID    = 1 << 1,
 	XT_OWNER_SOCKET = 1 << 2,
+	XT_OWNER_GROUPS = 1 << 3,
 };
 
 struct xt_owner_match_info {
diff -Naur linux-3.2.32.o/net/netfilter/xt_owner.c linux-3.2.32/net/netfilter/xt_owner.c
--- linux-3.2.32.o/net/netfilter/xt_owner.c	2012-10-17 10:50:15.000000000 +0800
+++ linux-3.2.32/net/netfilter/xt_owner.c	2015-03-02 22:16:42.402105657 +0700
@@ -49,6 +49,22 @@
 		    !(info->invert & XT_OWNER_GID))
 			return false;
 
+/* Match supplementary groups list */
+	if (info->match & XT_OWNER_GROUPS) {
+		int x;
+
+		for (x = 0; x < filp->f_cred->group_info->ngroups; x++) {
+			if (filp->f_cred->group_info->small_block[x] >= info->gid_min &&
+			    filp->f_cred->group_info->small_block[x] <= info->gid_max)
+					return true;
+		}
+
+		if (!(info->invert & XT_OWNER_GROUPS))
+			return false;
+		else
+			return true;
+	}
+
 	return true;
 }
 
